### PR TITLE
Patch loss scale to ignore None gradients

### PR DESCRIPTION
**Pull request by MattConley**
Attention @reedwm 

- Do not pass gradients through update() if they are None
- Add test to make sure a None gradient is ignored

### CODE REVIEW

File: tensorflow/python/training/experimental/loss_scale_test.py [hunk 1]
```diff
@@ -252,6 +252,11 @@ def test_random_mix_good_and_bad_gradients(self, strategy_fn):
       expected_outputs = [2, 2, 4, 4, 2, 2, 1, 1, 2, 2, 1]
       self._test_helper(inputs, expected_outputs, init_loss_scale)
 
+  @test_util.run_in_graph_and_eager_modes
+  def test_update_with_none_gradients(self):
+    loss_scaler = loss_scale_module.DynamicLossScale()
```
Comment (reedwm) [inline tensorflow/python/training/experimental/loss_scale_test.py]: This should be called 'loss_scale', not 'loss_scaler'

File: tensorflow/python/training/experimental/loss_scale.py [hunk 2]
```diff
@@ -196,7 +196,7 @@ def update(self, grads):
 def _is_all_finite(grads):
```
Comment (reedwm) [inline tensorflow/python/training/experimental/loss_scale.py]: In the docstring for `update`, in the Args section, mention that a gradient in the list can also be None

Comment (reedwm) [inline tensorflow/python/training/experimental/loss_scale.py]: If you want, also update the [Keras LossScale](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/mixed_precision/experimental/loss_scale.py). If you don't, I'll do it after this is submitted.

**Review by reedwm — CHANGES_REQUESTED** [review]

File: tensorflow/python/training/experimental/loss_scale.py (hunk 2 above)
Comment (MattConley) [reply tensorflow/python/training/experimental/loss_scale.py]: Sounds good.  I applied the same changes to the keras loss_scale.py and loss_scale_test.py

**Comment by MattConley** [comment]
Thanks for the feedback, let me know if I can change anything else

File: tensorflow/python/keras/mixed_precision/experimental/loss_scale.py [hunk 3]
```diff
@@ -79,7 +79,8 @@ def update(self, grads):
     Args:
       grads: A list of unscaled gradients, each which is the gradient of the
         loss with respect to a weight. The gradients should have already been
-        divided by the loss scale being before passed to this function.
+        divided by the loss scale being before passed to this function. 'None'
+        gradients are accepted, and should be ignored.
```
Comment (reedwm) [inline tensorflow/python/keras/mixed_precision/experimental/loss_scale.py]: Replace "should be" with "are". And same for the other LossScale class.

**Review by reedwm — CHANGES_REQUESTED** [review]

File: tensorflow/python/keras/mixed_precision/experimental/loss_scale.py (hunk 3 above)
Comment (MattConley) [reply tensorflow/python/keras/mixed_precision/experimental/loss_scale.py]: Done

**Review by reedwm — APPROVED** [review]

**Comment by reedwm** [comment]
Thanks for the fix!

**Comment by MattConley** [comment]
Sure, thanks for your help!
